src/main/java/org/jabref/gui/util/Theme.java

```diff
@@ -127,11 +249,35 @@ private void addAndWatchForChanges(Scene scene, URL cssFile, FileUpdateMonitor f
         }
     }
 
+    /**
+     * @return the Theme type
+     */
     public Type getType() {
         return type;
     }
 
-    public Path getPath() {
-        return pathToCss;
+    /**
+     * Provides the raw, configured custom CSS location. This should be a file system path, but the raw string is
+     * returned even if it is not valid in some way. For this reason, the main use case for this getter is to
+     * storing or display the user preference, rather than to read and use the CSS file.
+     *
+     * @return the raw configured CSS location
+     */
+    public String getCssPathString() {
+        return cssPathString;
+    }
```

Comment:
Hi @docrjp , I took a closer look into your changes.

I really did not get why you were removing the Path object in your changes.
The Path object in the `java.nio.file` package is just providing this: a simple Path, which is file system independent by design, not linked to a concrete file, but rather just the Path. Abstraction is a major difference between the old-style java io and the nio package. You can test the validity of the Path with Files.exists().

If you are unsure, whether a syntactically correct path is delivered by the PreferencesService: It should deliver a Path optional. I think this was not implemented before, but this would be a good addition. (I'm still in progress of refactoring JabRefPreferences and its surroundings. If possible, PreferencesService should always deliver high abstraction objects instead of raw strings - see ColumnPreferences for example)

I believe using raw Strings here is unnecessary and somewhat inconvenient.

Reply:
Digging even deeper, I realized it's still more complicated.

Currently JabRefPreferences is providing the Theme itself. I'm not yet sure, if it should...

Reply:
I thought a little about this. Besides that our complete theme architecture desires some attention and love, I think that keeping three different references in the class will provoke sooner or later unexpected bugs. The best would probably just to work inside the class with URLs and abolish the Path object, only return it, if asked by the prefs.

Reply:
I don't like it either! But it is making true on design decisions already in place both upstream and downstream of the getter.

**Every** use of the Path getter was immediately converted to a String i.e. throwing away the abstraction.

The constructor of Theme took a String already, passing in magic values of the path for the LIGHT and DARK themes.

In the case of the LIGHT theme, the magic value is an empty string, and therefore an empty Path, which can never be used to open a regular file at all, it just happens to be OK to have an empty Path on Windows/Linux/Mac file systems. Usually an empty path means "current working directory" -- that has no meaningful relationship to the LIGHT theme.

For the DARK theme, there is also some magic. 'Dark.css' is passed to the constructor, it then works backwards: by detecting that the URL is a resource, it guesses that it is the DARK theme, since this is the only theme at present that is stored in the jar. For a binary build, this will never exist, as even though it happens to be a valid Path, it is is never intended to be used as a file system Path. It implies something that isn't true.

Converting this string always to Path is therefore a misleading abstraction. It really fooled me, when I first came to it.

Related to this, is that I didn't want to second guess how the preference service might interact with the "path" string (as I say, it throws away the high level abstraction). Since there was no check for Path validity before, it might well be assuming that the getter always returns a value which when it converts back to a string, reflects the constructor, Changing that to Optional is will make that no longer be true. That could create new bugs, e.g. the path configuration being "lost", at the surprise of the user. It is a behavioural change to introduce Optional to this getter.

To really use the Path consistently as an abstraction, I'd really have to rip up the Theme class, maybe with a factory pattern and removing these magic constructor values, and also look at changing how the preferences uses Theme. That makes the change much bigger and may better be in a separate PR.

Reply:
How about this plan:

I will create two new issues, for follow up PRs:

- add more built in themes alongside Dark, e.g. a high contrast Dark theme for enhanced accessibility. In this we will eliminate the magic handling specific to Light and Dark theme, allowing for eaisier extensibility to built in themes
- add validation of user input of custom theme file path. In this we will refactor how the Theme and preference interact in relation to paths.

Between these we should be able to get into a good state. Happy to give theme preferences some love!

Reply:
I have created issues #7322 and #7323 for follow up.